src/Workspaces/Core/Portable/Shared/Extensions/IAsyncEnumerableExtensions.cs

```diff
@@ -4,12 +4,26 @@
 
 using System.Collections.Generic;
 using System.Collections.Immutable;
+using System.Runtime.CompilerServices;
 using System.Threading;
+using System.Threading.Channels;
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.PooledObjects;
 
 namespace Microsoft.CodeAnalysis.Shared.Extensions
 {
+    internal static class AsyncEnumerable<T>
+    {
+        public static readonly IAsyncEnumerable<T> Empty = GetEmptyAsync();
+
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
+        private static async IAsyncEnumerable<T> GetEmptyAsync()
+        {
+            yield break;
+        }
+#pragma warning restore CS1998 // Async method lacks 'await' operators and will run synchronously
```

Comment:
this is the simplest way i can find to just make a singleon empty IAsyncEnumerable.  Would be nice to have this in teh BCL :)

Reply:
Is there any efficiency concerns by not doing this more directly? I agree anything else seems like a PITA though to do.

Reply:
i'm not sure what woudl be the "more directly" approach :)